src/components/xray/XrayExportSession.vue

```diff
@@ -0,0 +1,751 @@
+      let icon;
+      switch (description) {
+        case "EXECUTING":
+          icon = "mdi-play";
+          break;
+        case "PASSED":
+          icon = "mid-check";
+          break;
+        case "FAILED":
+          icon = "mdi-alert-circle";
+          break;
+        case "TO DO":
+        default:
+          icon = "mid-check";
+          break;
+      }
+
+      return icon;
+    },
+    resetResults(type) {
+      this.selectTestExecution = type === "testExecution";
+      this.selectTestRun = type === "testRun";
+      this.selectTest = type === "test";
+      this.addResult = type === "result";
+      this.testExecutionLoading = true;
+      this.testRunLoading = true;
+      this.testLoading = true;
+      this.resultLoading = true;
+    },
+    async showDialog() {
+      this.dialog = true;
+      this.resetResults("testExecution");
+      let first = true;
+      console.log("Clicked showDialog");
+
+      for (const [i, credential] of Object.entries(this.credentials)) {
+        if (credential[0].type === "xray") {
+          try {
+            const data = await xrayIntegrationHelper.fetchTestExecutions(
+              credential[0].accessToken
+            );
+
+            if (first) {
+              this.testExecutions = data.map((testExecution) => ({
+                ...testExecution,
+                credential_index: i,
+              }));
+            } else {
+              this.testExecutions.push(
+                ...data.map((testExecution) => ({
+                  ...testExecution,
+                  credential_index: i,
+                }))
+              );
+            }
+
+            this.testExecutionLoading = false;
+          } catch (error) {
+            this.testExecutionLoading = false;
+            this.snackBar.enabled = true;
+            this.snackBar.message = error.message
+              ? error.message
+              : this.$tc("message.api_error", 1);
+          }
+        }
+      }
+    },
+    async handleSelectTestExecution(item) {
+      this.resetResults("testRun");
+      console.log("Clicked showTestRuns");
+
+      try {
+        const data = await xrayIntegrationHelper.fetchTestRuns(
+          this.credentials.xray[0].accessToken,
+          item.issueId
+        );
+
+        this.testRuns = data.map((testRun) => ({
+          ...testRun,
+          credential_index: item.credential_index,
+        }));
+
+        this.testRunLoading = false;
+      } catch (error) {
+        this.testRunLoading = false;
+        this.snackBar.enabled = true;
+        this.snackBar.message = error.message
+          ? error.message
+          : this.$tc("message.api_error", 1);
+      }
+      //this.selectedItem = item;
+    },
+    async file2Base64(file) {
+      return new Promise((resolve, reject) => {
+        const reader = new FileReader();
+        reader.readAsDataURL(file);
+        reader.onload = () => resolve(reader.result?.toString() || "");
```

Comment:
In order to make the upload work without throwing an error, I had to add `btoa` here.

`reader.onload = () => resolve(btoa(reader.result?.toString()) || "");`

But this doesn't actually maintain the correct `mimeType`.  So if the file is a `jpg`, it reads it as an octet-stream.  We may need to change the mime data at the beginning of the file for this to work.

Reply:
I fixed this by removing the mimeType from the base64 data string.